src/metabase/api/table.clj

```diff
@@ -551,4 +552,24 @@
                 :file     (get-in raw-params ["file" :tempfile])
                 :action   ::upload/replace}))
 
+(defn- delete-csv!
+  "This helper function exists to make testing the DELETE /api/table/:id endpoint easier."
```

Comment:
interesting, why don't you use `mt/user-http-request` to test this API? That method actually calls to the jetty handler, so everything is on the same thread.

Reply:
Just copy pasted from the other CSV routes for consistency. On reflection I suspect they took this approach to simplify file uploading, so thanks I'll try this suggestion.

Reply:
I had to improve `user-http-request` to make file uploading work when working on serialization API, so right now it could be possible to rework all tests to use it — if this makes sense, of course.

Reply:
This is the second time you've made that suggestion Sanya so this is the last time I'm allowed to say "next time" 😅

Reply:
Ended up making this change when the route moved to EE 😄